backend/app/resources/validate_utils.py

```diff
@@ -100,3 +102,67 @@ def validate_userinfo(userinfo, error_list):
             )
 
     return error_list
+
+
+def validate_user(user, user_str, error_list):
+    if not isinstance(user, dict):
+        error_list.append(f"The {user_str} supplied is not a dict.")
+        return error_list
+
+    if "auth_id" not in user:
+        error_list.append(f'The {user_str} supplied does not have field "auth_id".')
+
+    validate_userinfo(user["info"], error_list)
+
+
+def validate_meal_info(meal_info, error_list):
+    meal_info_fields = ["portions", "dietary_restrictions", "meal_suggestions"]
+
+    if not isinstance(meal_info, dict):
+        error_list.append("The meal_info info supplied is not a dict.")
+        return error_list
+
+    for key, val in meal_info.items():
+        if key not in meal_info_fields:
+            error_list.append(f'The meal_info info supplied has invalid field "{key}".')
+        elif key == "portions":
+            if type(val) is not int:
+                error_list.append("The portions supplied is not an int.")
+            if val <= 0:
+                error_list.append("The portions supplied must be greater than zero.")
+        elif key == "dietary_restrictions" and type(val) is not str:
+            error_list.append("The dietary_restrictions supplied is not a string.")
+        elif key == "meal_suggestions" and type(val) is not str:
+            error_list.append("The meal_suggestions supplied is not a string.")
+
+
+def validate_donation_info(donation_info, error_list):
+    donation_info_fields = [
+        "donor",
+        "commitment_date",
+        "meal_description",
+        "additional_info",
+    ]
+
+    if not isinstance(donation_info, dict):
+        error_list.append("The donation_info info supplied is not a dict.")
+        return error_list
+
+    for key, val in donation_info.items():
+        if key not in donation_info_fields:
+            error_list.append(
+                f'The donation_info info supplied has invalid field "{key}".'
+            )
+        elif key == "donor":
+            validate_user(val, "donation_info.donor", error_list)
```

Comment:
suggestion: maybe we should we also validate that val.role == "Donor" so if we ever somehow accidentally assign an ASP to a meal request it will fail.

Reply:
Thanks for the suggestion, just added this change 👍